src/scripts/main.js

```diff
@@ -1,3 +1,246 @@
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+
+    <label>Age:
+      <input name="age" data-qa="age" type="number">
+    </label>
+
+    <label>Salary:
+      <input name="salary" data-qa="salary" type="number">
+    </label>
+
+    <button>Save to table</button>
+  </form>
+`);
+
+const form = document.querySelector('form');
+
+form.addEventListener('submit', e => {
+  const data = new FormData(form);
+
+  e.preventDefault();
+
+  const age = +data.get('age');
+  const salary = +data.get('salary');
+
+  switch (true) {
+    case (!data.get('name')
+      || !data.get('position')
+      || !data.get('office')
+      || !age
+      || !salary):
+      // e.preventDefault();
+
+      pushNotification(10, 10, 'Fill in all fields',
+        'Necessary to fill all fields', 'error');
+
+      return;
+
+    case (data.get('name').length < 4):
+      // e.preventDefault();
+
+      pushNotification(10, 10, 'Too short Name',
+        'Enter a longer Name. Longer 3 characters', 'error');
+
+      return;
+
+    case (age < 18):
+      // e.preventDefault();
+
+      pushNotification(10, 10, 'Too young Age',
+        'You are too young for this shit.\n Min Age is 18', 'error');
+
+      return;
+
+    case (age > 90):
+      // e.preventDefault();
+
+      pushNotification(10, 10, 'Too old Age',
+        'You are too old for this shit.\n Max age is 90', 'error');
+
+      return;
+
+    default:
+      break;
+  }
+
+  tbody.insertAdjacentHTML('beforeend', `
+  <tr>
+    <td>${data.get('name')}</td>
+    <td>${data.get('position')}</td>
+    <td>${data.get('office')}</td>
+    <td>${age}</td>
+    <td>$${salary.toLocaleString()}</td>
+  </tr>
+  `);
+
+  pushNotification(10, 10, 'Person was appended',
+    'There are even more of us', 'success');
+
+  updateVariables();
+});
+
+//  dblclick on cell
+tbody.addEventListener('dblclick', e => {
+  const cell = e.target;
+
+  if (!cell.matches('td')) {
+    return;
+  }
+
+  const initialText = cell.innerText;
+
+  cell.innerText = '';
+
+  cell.insertAdjacentHTML('beforeend', `
+    <input class="cell-input" value="${initialText}">
+  `);
+
+  const inputForEdit = cell.lastElementChild;
+
+  inputForEdit.focus();
+
+  extraEventsForInput(cell, inputForEdit, initialText);
+
+  updateVariables();
+});
+
+function extraEventsForInput(target, input, initialValue) {
+  //  blur
+  input.addEventListener('blur', () => {
+    removeAndAdd(target, input, initialValue);
+  });
+
+  //  keyup
+  input.addEventListener('keyup', e => {
+    if (e.code === 'Enter') {
+      removeAndAdd(target, input, initialValue);
+    }
+  });
+};
+
+function removeAndAdd(target, input, initialValue) {
+  const value = (input.value) ? input.value : initialValue;
+  const inputInside = input;
+
+  if (inputInside) {
+    inputInside.remove();
+  }
+
+  target.innerText = value;
+};
+
+function updateVariables() {
```

Comment:
Consider wrapping the code updating variables in a function to avoid repetition and improve readability.